src/solvers/smt2/smt2_dec.cpp

```diff
@@ -44,22 +46,24 @@ std::string smt2_dect::decision_procedure_text() const
 
 smt2_temp_filet::smt2_temp_filet()
 {
-  temp_out_filename=get_temporary_file("smt2_dec_out_", "");
+  temp_problem_filename = get_temporary_file("smt2_dec_problem_", "");
 
   temp_out.open(
-    temp_out_filename.c_str(),
-    std::ios_base::out | std::ios_base::trunc);
+    temp_problem_filename.c_str(), std::ios_base::out | std::ios_base::trunc);
 }
 
 smt2_temp_filet::~smt2_temp_filet()
 {
   temp_out.close();
 
-  if(temp_out_filename!="")
-    unlink(temp_out_filename.c_str());
+  if(temp_problem_filename != "")
+    unlink(temp_problem_filename.c_str());
+
+  if(temp_stdout_filename != "")
+    unlink(temp_stdout_filename.c_str());
 
-  if(temp_result_filename!="")
-    unlink(temp_result_filename.c_str());
+  if(temp_stderr_filename != "")
+    unlink(temp_stderr_filename.c_str());
```

Comment:
Can we actually get rid of `smt2_temp_filet` completely? Back when this code was first written we probably hadn't yet got `temporary_filet` - we now do, and that should be the only place managing temporary files. If for some reason this is infeasible then surely `temp_dirt` would work? But all of this should be separate commit not to distract us from the more trivial `system` -> `run` transition.

Reply:
Yes, I did contemplate this; will look into it.